### PR TITLE
[CLS-10980] allow the customer to specify cpu/memory for the agent container

### PR DESCRIPTION
**Pull request by arields1**

### CODE REVIEW

File: charts/s1-agent/templates/agent/daemonset.yaml [hunk 1]
```diff
@@ -36,7 +36,11 @@ spec:
         image: "{{ include "agent.full_url" . }}"
         imagePullPolicy: {{ default "IfNotPresent" .Values.configuration.imagePullPolicy }}
         resources:
+{{- if .Values.configuration.env.injection.enabled }}
```
[the remark this answers is not in the document]
Comment (arields1) [reply charts/s1-agent/templates/agent/daemonset.yaml:39]: Fixed

File: charts/s1-agent/values.yaml [hunk 2]
```diff
@@ -67,6 +67,13 @@ agentInjection:
     namespaceSelector:
       matchLabels:
         agent-injection-enabled: "true"
+  resources:
+    limits:
+      cpu: 900m
+      memory: 1945Mi # Almost equals to 1.9Gi but isn't fractional
+    requests:
+      cpu: 100m
+      memory: 100Mi
```
[the remark this answers is not in the document]
Comment (arields1) [reply charts/s1-agent/values.yaml:76]: @natishauli @oded-s1 do you guys have suggestions?

[the remark this answers is not in the document]
Comment (natishauli) [reply charts/s1-agent/values.yaml:76]: yes, I suggest to set limits
cpu: 500m
memory: 500Mi

File: charts/s1-agent/templates/agent/daemonset.yaml [hunk 3]
```diff
@@ -36,7 +36,11 @@ spec:
...
+{{ toYaml .Values.agentInjection.resources | indent 10 }}
+{{- else }}
```
Comment (oded-s1) [inline charts/s1-agent/templates/agent/daemonset.yaml:41]: what will happen in hybrid cluster?

Comment (danisevas) [reply charts/s1-agent/templates/agent/daemonset.yaml:41]: @arields1
 that is an excellent point!
The daemonset is used only on non-fargate nodes, so it should be the resources specified in the agent section.
the injected agent takes its config from the configmap, so there you should take the injected section

**Review by danisevas — APPROVED** [review]

**Comment by arields1** [comment]
Automation test is good

**Pull request merged**